network-mux/src/Network/Mux/Bearer/Pipe.hs

```diff
@@ -59,6 +59,7 @@ pipeAsMuxBearer tracer pcRead pcWrite = do
                   traceWith tracer $ Mx.MuxTraceRecvPayloadStart (fromIntegral $ Mx.msLength header)
                   blob <- recvLen' pcRead (fromIntegral $ Mx.msLength header) []
                   ts <- getMonotonicTime
+                  traceWith tracer (Mx.MuxTraceRecvDeltaQObservation header ts)
```

Comment:
Seems to me that these traces should be in the generic implementation or use of a bearer, and not have to be in each bearer implementation separately.